client/src/main/java/org/apache/cassandra/sidecar/client/RequestExecutor.java

```diff
@@ -305,26 +267,21 @@ private void applyRetryPolicy(CompletableFuture<HttpResponse> future,
         context.retryPolicy()
                .onResponse(future, request, response, throwable, attempt, retryOnNewHost, (nextAttempt, delay) -> {
             String statusCode = response != null ? String.valueOf(response.statusCode()) : "<Not Available>";
+            SidecarInstance nextInstance = sidecarInstance;
             if (iterator.hasNext())
             {
-                if (response == null || response.statusCode() != HttpResponseStatus.ACCEPTED.code())
-                {
-                    logger.warn("Retrying request on next instance after {}ms. Failed on instance={}, " +
-                                "attempt={}, statusCode={}, request={}", delay, sidecarInstance, attempt, statusCode,
-                                request, throwable);
-                }
-                schedule(delay, () -> executeWithRetries(future, iterator, context, nextAttempt, throwable));
+                nextInstance = iterator.next();
             }
-            else
+
+            if (response == null || response.statusCode() != HttpResponseStatus.ACCEPTED.code())
             {
-                if (response == null || response.statusCode() != HttpResponseStatus.ACCEPTED.code())
-                {
-                    logger.warn("Retrying request on same instance after {}ms. Failed on instance={}, " +
-                                "attempt={}, statusCode={}, request={}", delay, sidecarInstance, attempt, statusCode,
-                                request, throwable);
-                }
-                schedule(delay, () -> executeWithRetries(future, iterator, sidecarInstance, context, nextAttempt));
+                logger.warn("Retrying request on {} instance after {}ms. " +
+                            "Failed on instance={}, attempt={}, statusCode={}",
+                            nextInstance == sidecarInstance ? "same" : "next", delay,
+                            nextInstance, attempt, statusCode, throwable);
```

Comment:
Wouldn't the failed instance be `sidecarInstance` instead?
```suggestion
                            sidecarInstance, attempt, statusCode, throwable);
```

Reply:
good catch